metaflow/plugins/aws/batch/batch.py

```diff
@@ -53,15 +54,26 @@ def _search_jobs(self, flow_name, run_id, user):
         return jobs
 
     def _job_name(self, user, flow_name, run_id, step_name, task_id, retry_count):
-        return '{user}-{flow_name}-{run_id}-{step_name}-{task_id}-{retry_count}'.format(
-            user=user,
+        job_name = '{user}-{flow_name}-{run_id}-{step_name}-{task_id}-{retry_count}'.format(
+            user=user.replace(" ", ""),
```

Comment:
This will cause issues for users `a bc` and `ab c`. Instead of removing whitespace, we should disallow it as mentioned in the comment below.